### PR TITLE
Fix a small performance issue with `GET /api/v1/applications` endpoint

### PR DESCRIPTION
**Pull request by steventux**
## Context

We've not included a couple of associated models as includes in the query the `GET /applications` endpoint uses to fetch applications. This was causing _many_ n+1 queries like

```
ApplicationReference Load (0.6ms)  SELECT "references".* FROM "references" WHERE "references"."application_form_id" = $1 AND "references"."feedback_status" = $2 AND "references"."selected" = $3 ORDER BY id ASC  [["application_form_id", 186], ["feedback_status", "feedback_provided"], ["selected", true]]       
↳ app/presenters/vendor_api/single_application_presenter.rb:183:in `map'                              
ApplicationReference Load (0.6ms)  SELECT "references".* FROM "references" WHERE "references"."application_form_id" = $1 AND "references"."feedback_status" = $2 AND "references"."selected" = $3 ORDER BY id ASC  [["application_form_id", 184], ["feedback_status", "feedback_provided"], ["selected", true]]
↳ app/presenters/vendor_api/single_application_presenter.rb:183:in `map'
```

## Changes proposed in this pull request

Adds `application_form: [:application_references]` and `offer: [:conditions]` includes the the API version of the query.

## Link to Trello card

Noticed this when investigating:
https://trello.com/c/yzKDiTtI/4037-spike-investigate-performance-issues-with-get-applications-endpoint

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/presenters/vendor_api/single_application_presenter.rb [hunk 1]
```diff
@@ -180,7 +180,9 @@ def experience_description(experience)
     end
 
     def references
-      application_form.application_references.selected.map do |reference|
+      application_form.application_references.map do |reference|
+        next unless reference.selected && reference.feedback_status == 'feedback_provided'
```
Comment (cwrw) [inline app/presenters/vendor_api/single_application_presenter.rb:184]: I don't thinks we need `reference.feedback_status == 'feedback_provided'` as that is part of the selected scope: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_reference.rb#L13

Comment (cwrw) [reply app/presenters/vendor_api/single_application_presenter.rb:184]: we might need to compact references to avoid `nil` values being in the array

**Review by cwrw — APPROVED** [review]
Looks good, some minor comments inline

File: app/presenters/vendor_api/single_application_presenter.rb (hunk 1 above)
Comment (steventux) [reply app/presenters/vendor_api/single_application_presenter.rb]: Thanks @cwrw for review, I found that using the `.selected` scope was what triggered the multiple queries. There's mention of this problem elsewhere in the presenter https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/presenters/vendor_api/single_application_presenter.rb#L225-L227

**Review by cwrw — APPROVED** [review]

**Pull request merged**